<a href="https://colab.research.google.com/github/Umair01-code/FlyRank-ML-Intrenship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Umair01-code/FlyRank-ML-Intrenship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

# 1. Refresh / Content Opportunity Scoring

I chose Lane 2: Refresh / Content Opportunity Scoring because it focuses on helping content teams decide which pages should be reviewed first. Large websites often contain thousands of pages, making manual review difficult and time-consuming.

From the starter notebooks, I learned that observable signals such as impressions, CTR, page position, and content freshness can help prioritize pages for review. I want to explore how these signals can be combined into an evidence-based ranking that supports human decision-making rather than replacing it.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

# 2. Decision, Action, Cost of a Wrong Call

### Research Question

Which content pages should be reviewed first for refresh based on observable search and engagement signals?

### Decision

My work aims to improve the decision of which pages should be prioritized for review instead of checking every page manually.

### Action

A content editor or SEO specialist can use the ranked list to decide which pages to refresh, expand, monitor, or investigate first.

### Cost of a Wrong Recommendation

If an important page is ranked too low, the team may miss an opportunity to improve valuable content. If a low-priority page is ranked too high, the team may spend time reviewing content that provides little benefit. Therefore, the ranking should support human decisions rather than replace them.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [18]:
import pandas as pd

# Load the starter dataset
df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Total Pages:", len(df))
print("Total Columns:", len(df.columns))
print("Average CTR:", round(df["ctr"].mean(), 3))
print("Average Position:", round(df["avg_position"].mean(), 2))
print("Percentage of Declining Pages:",
      round((df["trend_direction"] == "down").mean() * 100, 2), "%")

Total Pages: 30000
Total Columns: 44
Average CTR: 0.511
Average Position: 16.34
Percentage of Declining Pages: 54.21 %


### Observations

The starter dataset contains 30,000 pages, which provides enough data to analyze content performance patterns. Around 54.21% of the pages are marked as declining, indicating there are many potential candidates for content review. The average CTR is 0.511, suggesting there are opportunities to study which factors are associated with better click-through rates. These observations make the Refresh / Content Opportunity Scoring lane a practical choice for the internship.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

# 4. What I Can and Can't Claim

This project will use observable search and engagement signals to support content review decisions. The findings should be interpreted as observed patterns from the anonymized dataset rather than universal rules.

The ranked recommendations are intended to help prioritize pages for human review. They do not guarantee that refreshing a page will improve search performance.

This project cannot prove how Google's ranking algorithm works, demonstrate causal relationships, or guarantee future traffic improvements.



## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.